src/models/identification/useIdConfigSchemas.js

```diff
@@ -34,23 +36,63 @@ export default function useIdConfigSchemas() {
     const siteSpecies = get(data, ['site.species', 'value'], []);
     const siteItisIds = siteSpecies.map(species => species.itisTsn);
 
-    const allAlgorithms = flatMap(
+    const allAlgorithms = reduce(
       Object.values(detectionConfig),
-      model =>
-        flatMap(get(model, 'supported_species', []), species => {
-          if (!siteItisIds.includes(species?.itis_id)) return [];
-          const idAlgorithmObject = species?.id_algos || [];
-          const idAlgorithms = map(
-            idAlgorithmObject,
-            (algorithmSchema, algorithmKey) => ({
-              label: algorithmSchema?.description,
-              value: algorithmKey,
-              taxonomy: species?.scientific_name,
-              itisId: species?.itis_id,
-            }),
+      (memo, model) => {
+        const supportedSpecies = model?.supported_species;
+        const speciesMap = {};
+        const idAlgos = flatMap(supportedSpecies, singleSpecies => {
+          const commonName =
+            singleSpecies?.common_name ||
+            intl.formatMessage({ id: 'UNKNOWN_COMMON_NAME' });
+          const speciesName =
+            singleSpecies?.scientific_name ||
+            intl.formatMessage({ id: 'UNKNOWN_SCIENTIFIC_NAME' });
+          if (!siteItisIds.includes(singleSpecies?.itis_id))
+            return [];
+          const safeSingleSpeciesIdAlgos =
+            singleSpecies?.id_algos || [];
+          const algoChoices = map(
+            safeSingleSpeciesIdAlgos,
+            (idAlgo, key) => {
+              // populate the species map with species name arrays keyed by algorithm key so that they can later be joined and added to the final label
+              const currentSpecies = get(speciesMap, key, []);
+              if (currentSpecies.length > 0) {
+                speciesMap[key].push(
+                  commonName + ' (' + speciesName + ')',
+                );
+              } else {
+                speciesMap[key] = [
+                  commonName + ' (' + speciesName + ')',
+                ];
+              }
+
+              const algoDescription =
+                idAlgo?.description ||
+                intl.formatMessage({
+                  id: 'UNKNOWN_ALGORITHM_DESCRIPTION',
+                });
+              return {
+                value: key,
+                label: algoDescription + ' for: ',
+              };
+            },
           );
-          return idAlgorithms;
-        }),
+
+          return algoChoices;
+        });
+
+        const idAlgosWithImprovedLabels = map(idAlgos, idAlgo => {
+          const baseLabel = get(idAlgo, 'label', '');
+          const labelsToAdd = get(speciesMap, [idAlgo?.value], '');
+          return {
+            ...idAlgo,
+            label: baseLabel + labelsToAdd.join(', '),
```

Comment:
Done in https://github.com/WildMeOrg/codex-frontend/pull/468/commits/586560061c25873d532a8129d1061fa5002b41ac. Note that I still have to keep baseLabel out of the array (i.e., I couldn't pass [baseLabel, ...labelsToAdd] to formatList) because it would treat baseLabel as a valid thing in the list in terms of conjunction. In other words, it would say things like, "Hotspotter for: and wolf...". The "and" there is a bug.